src/transport_portals4.h

```diff
@@ -650,6 +652,30 @@ shmem_transport_put_nbi(shmem_transport_ctx_t* ctx, void *target, const void *so
 #endif
 }
 
+
+static inline
+void
+shmem_transport_put_nb(shmem_transport_ctx_t* ctx, void *target, const void *source, size_t len,
+                                int pe, long *completion)
+{
+    if (ctx->options & SHMEMX_CTX_ISOLATE_PUTS) {
+#ifdef ENABLE_REMOTE_VIRTUAL_ADDRESSING
+        shmem_transport_portals4_put_nb_internal(ctx, target, source, len, pe,
+                                                 completion,
+                                                 shmem_transport_portals4_pt,
+                                                 -1);
+#else
+        shmem_transport_portals4_put_nb_internal(ctx, target, source, len, pe,
+                                                 completion,
+                                                 shmem_transport_portals4_data_pt,
+                                                 shmem_transport_portals4_heap_pt);
+#endif
+    } else {
+        shmem_transport_put_nbi(ctx, target, source, len, pe);
+    }
+}
+
```

Comment:
So if I understand correctly, only the default context can do bounce buffering and the "frag" stuff.  In other words, it won't be optional for user-created contexts to do bounce buffering.  It's no problem, just making sure that's the design choice.

Reply:
Right.  The user can enable the ```SHMEMX_CTX_ISOLATE_PUT``` and ```SHMEMX_CTX_BOUNCE_BUFFER``` on a given context if they want, but they are disabled by default.  We should probably have a unit test for this, come to think of it.

Reply:
If a user only sets `SHMEMX_CTX_BOUNCE_BUFFER` (and not `ISOLATE_PUT`), it won't do bounce buffering, right?  Should there be a warning or something for that case?

Reply:
Oh!  I didn't notice that ```shmem_transport_put_nbi``` doesn't support bounce buffering.  In that case, maybe I shouldn't try to separate these features -- in Portals, ```SHMEMX_CTX_BOUNCE_BUFFER``` enables both optimizations.  Sound ok?

Reply:
Yup!

Reply:
Done.